source/extensions/filters/network/zookeeper_proxy/zookeeper_utils.cc

```diff
@@ -0,0 +1,41 @@
+#include "extensions/filters/network/zookeeper_proxy/zookeeper_utils.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace ZooKeeperProxy {
+
+void BufferHelper::peekInt32(Buffer::Instance& buffer, uint64_t& offset, int32_t& val) {
+  val = buffer.peekBEInt<int32_t>(offset);
+  offset += sizeof(int32_t);
+}
+
+void BufferHelper::peekInt64(Buffer::Instance& buffer, uint64_t& offset, int64_t& val) {
```

Comment:
Now that you're using exceptions for error handling, wdyt about returning the parsed value instead of using out params? I think it might make the calling code easier to read?

Reply:
Yeah I like that idea.